src/NexusMods.CLI/CliGuidedInstaller.cs

```diff
@@ -0,0 +1,244 @@
+
+    private void RenderTableOfOptions(
+        OptionGroup group,
+        IReadOnlyCollection<SelectedOption> selectedOptions)
+    {
+        var key = 1;
+        var row = group.Options
+            .Select(option =>
+            {
+                var hasSelected = selectedOptions.Any(x => x.OptionId == option.Id);
+
+                return new object[]
+                {
+                    key++,
+                    RenderOptionState(hasSelected, option.Type),
+                    option.Name,
+                    option.Description
+                };
+            })
+            .Append(TableOfOptionsFooterBackToGroupSelection)
+            .Append(TableOfOptionsFooterCancel);
+
+        var table = new Table(TableOfOptionsHeaders, row.ToArray(), group.Description);
+        Renderer.Render(table);
+    }
+
+    private static string RenderOptionState(bool hasSelected, OptionType type)
+    {
+        if (hasSelected) return "Selected";
+
+        Debug.Assert(Enum.IsDefined(typeof(OptionType), type));
+
+        return type switch
+        {
+            OptionType.Disabled => "Disabled",
+            // NOTE (erri120): These are the "initial" states.
+            // The user can toggle a pre-selected option, which
+            // won't change the type but the "hasSelected" value.
+            OptionType.Available or OptionType.PreSelected => "Off",
+            OptionType.Required => "Required",
+            _ => throw new UnreachableException($"hasSelected: {hasSelected}, type: {type}")
+        };
+    }
+
+    private static void UpdatedSelectedGroup(
+        OptionGroup currentGroup,
+        ICollection<SelectedOption> selectedOptions,
+        string input)
+    {
+        var optionIndex = ParseNumericalUserInput(input, currentGroup.Options.Length);
+        if (optionIndex < 0) return;
+
+        var targetOption = currentGroup.Options[optionIndex];
+
+        // can't toggle disabled or required options
+        if (targetOption.Type is OptionType.Disabled or OptionType.Required) return;
+
+        var hasSelected = selectedOptions.TryGetFirst(x => x.OptionId == targetOption.Id, out var selectedOption);
+        if (hasSelected)
+        {
+            // the target option is already selected, the user wants to "toggle" it
+            selectedOptions.Remove(selectedOption);
+            return;
+        }
+
+        // the target option is not selected, the user wants to select it
+        if (currentGroup.Type is OptionGroupType.ExactlyOne or OptionGroupType.AtMostOne)
+        {
+            // "deselect" everything
+            selectedOptions.Clear();
+        }
+
+        selectedOptions.Add(new SelectedOption(currentGroup.Id, targetOption.Id));
+    }
+
+    private static string GetUserInput()
+    {
+        return (Console.ReadLine() ?? "").Trim();
+    }
+
+    private static int ParseNumericalUserInput(string input, int upperLimit)
+    {
+        try
+        {
+            var idx = int.Parse(input) - 1;
```

Comment:
Add comment to explain the -1
It might be missed at a glance and cause confusion